### DISCUSSION TITLE
Shared & StringProtocol conformance

### DISCUSSION THREAD

**Discussion by easiwriter** [pointfreeco/swift-composable-architecture#3513] [Q&A]
I have the following:

<img width="886" alt="Screenshot 2024-11-29 at 13 54 13" src="https://github.com/user-attachments/assets/778b2e33-3283-4077-b201-dc76c1b5b838">

and:

<img width="1172" alt="Screenshot 2024-11-29 at 13 53 36" src="https://github.com/user-attachments/assets/7f9f492b-383a-4eba-8519-1eb08fb49f7c">

I'm afraid I don't understand why I'm getting this error (name is a String).

**Comment by Alex293** [comment]
You pass $project to the ForEch which provides you with a binding in the closure. Text don’t take a binding but a string. You can drop the $ or use project.wrappedValue

**Reply by pyrtsa** [reply]
That's a valid option too, if the view only needs read access to the shared values.

**Comment by pyrtsa** [comment]
I think you should name the local variable of the `ForEach` trailing closure `$project` rather than `project`. That should trigger the compiler magic to introduce `project` to refer to the `wrappedValue`.

**Reply by easiwriter** [reply]
Doh, how obvious. Thankyou.
